frame/finality-tracker/src/lib.rs

```diff
@@ -67,11 +77,11 @@ decl_module! {
 		/// Hint that the author of this block thinks the best finalized
 		/// block is the given number.
 		fn final_hint(origin, #[compact] hint: T::BlockNumber) {
-			ensure_none(origin)?;
-			assert!(!<Self as Store>::Update::exists(), "Final hint must be updated only once in the block");
-			assert!(
+			ensure_none(origin).map_err(|e| e.as_str())?;
+			ensure!(!<Self as Store>::Update::exists(), Error::AlreadyUpdated);
+			ensure!(
```

Comment:
@rphmeier FYI

Reply:
Updated `assert!` statements to `ensure!` statements within the dispatchable function.